sponsor-dapp-v2/src/views/ManagePositions.js

```diff
@@ -140,6 +146,18 @@ function useFinancialContractData(tokenAddress) {
     }
   ];
 
+  // The user must be the token sponsor, the contract must be in the live state (and not be ready to expire) for the
+  // manage buttons to be enabled.
+  data.canManagePosition =
+    data.isTokenSponsor && !data.frozen && data.derivativeStorage.state === ContractStateEnum.LIVE;
+
+  // There are two cases where the redeem button should be enabled:
+  // 1. The contract is in a live (not ready to expire) state and the user is the token sponsor.
+  // 2. The contract is settled.
+  data.canRedeem =
+    (!data.frozen && data.isTokenSponsor && data.derivativeStorage.state === ContractStateEnum.LIVE) ||
```

Comment:
Is the first part just data.canManagePosition?

Also, do we want to allow withdrawals from settled?

Reply:
Both good points, yes and yes.

Reply:
Done. PTAL.